### ISSUE TITLE
Environment key "jest/globals" is unknown

### ISSUE THREAD

**Issue by Bardiamist** [typescript-eslint/typescript-eslint#5132]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  extends: [
    'plugin:@typescript-eslint/recommended',
    '@react-native-community',
    'airbnb',
  ],
  ...
};
```

### Expected Result

✨  Done in 26.63s.

### Actual Result

```
Oops! Something went wrong! :(

ESLint: 8.16.0

Error: .eslintrc.js » @react-native-community/eslint-config#overrides[2]:
        Environment key "jest/globals" is unknown

    at /Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2058:23
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateEnvironment (/Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2052:34)
    at ConfigValidator.validateConfigArray (/Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2202:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (/Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3947:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3753:21)
    at FileEnumerator._iterateFilesRecursive (/Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/eslint/lib/cli-engine/file-enumerator.js:446:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (/Users/bardiamist/Documents/my-proj/my-proj-mobile/node_modules/eslint/lib/cli-engine/file-enumerator.js:297:49)
    at iterateFiles.next (<anonymous>)
error Command failed with exit code 2.
```

### Additional Info

This error appeared after update "@typescript-eslint/eslint-plugin", "@typescript-eslint/parser",  and "@typescript-eslint/typescript-estree" from "5.24.0" to "5.25.0". Still exist in 5.26.0 and in 5.27.0.
So I still use 5.24.0 because it works without errors.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.27.0` |
| `@typescript-eslint/parser`            | `5.27.0` |
| `@typescript-eslint/typescript-estree` | `5.27.0` |
| `TypeScript`                           | `4.7.2` |
| `ESLint`                               | `8.16.0` |
| `node`                                 | `18.2.0` |
| `@react-native-community/eslint-config` | `3.0.2` |

**Comment by bradzacher** [comment]
> Error: .eslintrc.js » @react-native-community/eslint-config#overrides[2]:
>        Environment key "jest/globals" is unknown

From the error message, it's not an error to do with our project, sorry.

We don't do anything with jest publicly in our packages, so updating our version was unlikely to be the direct cause.

If you can create an isolated repro repo, might be able to help you out. But as it stands - closing as an external issue.

**Comment by Bardiamist** [comment]
Fixed in 5.27.1

**Comment by borisyankov** [comment]
I still encounter the same issue on `5.27.1`
